### PR TITLE
Dotnet masterclass week3

### PR DESCRIPTION
**Pull request by zmzade**

### CODE REVIEW

File: week3/Program.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+{
+    public string Name { get; set; } = "cow";
+    public string Sound { get; set; } = "maa maa";
+}
+
+class Animal
+{
+    public string MakeSound(IAnimal animal)
+    {
+        return $"{animal.Name} sags {animal.Sound}";
```
Comment (Vertygo) [inline week3/Program.cs]: small typo here `says` :)

File: week3/Program.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            }
+            return balance;
+        }
+    }
+    public Account(int amount)
+    {
+        Deposit(amount);
+    }
+
+    public List<Transaction> transactions = new List<Transaction>();
```
Comment (Vertygo) [inline week3/Program.cs]: I'm just curious why did you decide to have a list of transactions?

**Review by Vertygo — CHANGES_REQUESTED** [review]
Please add exercises from the class 3.

File: week3/Program.cs (hunk 2 above)
Comment (zmzade) [reply week3/Program.cs]: I could also create a private field `_balance`, return it by get method and then add or subtract from it by withdraw and deposit methods. It works and  balance is still read only. But I chose to create a list of objects from withdraw or deposit and then loop over them to get balance. According to exercise description, It is better to choose the first solution.

**Pull request merged**